controllers/services/credits.js

```diff
@@ -0,0 +1,146 @@
+/**
+* Copyright © 2017, ACM@UIUC
+*
+* This file is part of the Groot Project.  
+* 
+* The Groot Project is open source software, released under the University of
+* Illinois/NCSA Open Source License. You should have received a copy of
+* this license in a file with the distribution.
+**/
+
+const SERVICES_URL = process.env.SERVICES_URL || 'http://localhost:8000';
+const GROOT_ACCESS_TOKEN = process.env.GROOT_ACCESS_TOKEN || "TEMP_STRING";
+const STRIPE_PUBLISHABLE_KEY = process.env.STRIPE_PUBLISHABLE_KEY || "TEMP_STRING";
+const request = require('request');
+const moment = require('moment');
+const utils = require('../../etc/utils.js');
+
+module.exports = function(app){
+  app.get('/credits', function(req, res) {
+    if (!req.session.roles.isStudent) {
+      return res.redirect('/intranet');
+    }
+
+    request({
+      url: `${SERVICES_URL}/credits/transactions`,
+      method: "GET",
+      json: true,
+      headers: {
+        "Authorization": GROOT_ACCESS_TOKEN
+      },
+      qs: {
+        netid: req.session.student.netid
+      }
+    }, function(error, response, body) {
+      if (response && response.statusCode == 200) {
+        for(var t of body.transactions){
+          t.created_at = moment(t.created_at)
+            .format('MMMM Do YYYY, h:mm:ss a');
+        }
+        return res.render('credits', {
+          authenticated: true,
+          transactions: body.transactions,
+          balance: body.balance.toFixed(2),
+          messages: req.flash('success'),
+          errors: req.flash('error')
+        })
+      }
+      res.sendStatus(500);
+    });
+  });
+  app.get('/credits/purchaseMembership', function(req, res) {
+    // TODO: Redirect if not a 'pre-member'
```

Comment:
Can we integrate this as part of the join process?  Maybe have a "Pay now" option and "Pay in person" option?